services/templates/service.go

```diff
@@ -9,3 +29,65 @@ type service struct {
 func NewTemplateService() Service {
 	return &service{}
```

Comment:
Maybe instead of relying on `context` for invoking the storage and s3 methods, we can pass them as a dependency to this service struct and use them directly.